test/new-e2e/tests/security-agent-functional/security_agent_test.go

```diff
@@ -0,0 +1,98 @@
+	"flag"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/windows"
+	windowsCommon "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common"
+	windowsAgent "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/common/agent"
+	componentsos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/require"
+)
+
+type vmSuite struct {
+	e2e.BaseSuite[environments.Host]
+
+	testspath string
+}
+
+var (
+	devMode = flag.Bool("devmode", false, "run tests in dev mode")
+)
+
+func TestVMSuite(t *testing.T) {
+	suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(awshost.ProvisionerNoAgentNoFakeIntake(awshost.WithEC2InstanceOptions(ec2.WithOS(componentsos.WindowsDefault))))}
+	if *devMode {
+		suiteParams = append(suiteParams, e2e.WithDevMode())
+	}
+
+	e2e.Run(t, &vmSuite{}, suiteParams...)
+}
+
+func (v *vmSuite) SetupSuite() {
+	t := v.T()
+
+	// Get the absolute path to the test assets directory
+	currDir, err := os.Getwd()
+	require.NoError(t, err)
+
+	reporoot, _ := filepath.Abs(filepath.Join(currDir, "..", "..", "..", ".."))
+	kitchenDir := filepath.Join(reporoot, "test", "kitchen", "site-cookbooks")
+	v.testspath = filepath.Join(kitchenDir, "dd-security-agent-check", "files", "tests")
+}
+
+func (v *vmSuite) TestSystemProbeSuite() {
+	v.BaseSuite.SetupSuite()
+	t := v.T()
+	// get the remote host
+	vm := v.Env().RemoteHost
+
+	rs := windows.NewRemoteExecutable(vm, t, "testsuite.exe", v.testspath)
+	err := rs.FindTestPrograms()
+	require.NoError(t, err)
+
+	err = rs.CreateRemotePaths()
+	require.NoError(t, err)
+
+	err = rs.CopyFiles()
+	require.NoError(t, err)
+
+	// install the agent (just so we can get the driver(s) installed)
+	agentPackage, err := windowsAgent.GetPackageFromEnv()
+	require.NoError(t, err)
+	remoteMSIPath, err := windowsCommon.GetTemporaryFile(vm)
+	require.NoError(t, err)
+	t.Logf("Getting install package %s...", agentPackage.URL)
+	err = windowsCommon.PutOrDownloadFile(vm, agentPackage.URL, remoteMSIPath)
+	require.NoError(t, err)
+
+	err = windowsCommon.InstallMSI(vm, remoteMSIPath, "", "")
+	t.Log("Install complete")
+	require.NoError(t, err)
+
+	time.Sleep(30 * time.Second)
+	// disable the agent, and enable the drivers for testing
+	_, err = vm.Execute("stop-service -force datadogagent")
+	require.NoError(t, err)
+	_, err = vm.Execute("sc.exe config datadogagent start= disabled")
+	require.NoError(t, err)
+	_, err = vm.Execute("sc.exe config ddnpm start= demand")
+	require.NoError(t, err)
+	_, err = vm.Execute("start-service ddnpm")
+	require.NoError(t, err)
+
+	rs.RunTests("5m") // security agent tests can take a while waiting for ETW to start.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Can you add a probe to wait for ETW to be ready?

Reply:
No.   And this particular timeout is the argument to `go test`; it's making sure that `go test` will time out within 5m.  When running precompiled `go test` executables, by default there's no timeout at all; it will run indefinitely.

Reply:
Have you considered running the process with a `Cancel` context, and kill it on timeout?

Reply:
I don't think I explained this well.

The test takes a precompiled `go test` binary, copies to the VM, and executes it (in this case `.\testsuite.exe <args>`.
The execution of the executable on the remote host has no timeout by default (this is a "feature" of go test; when you compile a test into a standalone binary, instead of having a default timeout (20m), it runs indefinitely if the test hangs.

This particular line effectively adds the parameter `-test.timeout 5m` so that the executable on the remote host will not block indefinitely.